assimilation_code/modules/io/direct_netcdf_mod.f90

```diff
@@ -856,12 +878,28 @@ subroutine read_variables(ncfile_in, var_block, start_var, end_var, domain)
    dims = get_io_dim_lengths(domain, i)
 
    ret = nf90_inq_varid(ncfile_in, get_variable_name(domain, i), var_id)
-   call nc_check(ret, 'read_variables: nf90_inq_varid',trim(get_variable_name(domain,i)) )
+   call nc_check(ret, 'read_variables: nf90_inq_varid', varname)
+
+   ! check for unsupported features
+   ! In order to support them, we'd have to apply them when we write as well,
+   ! and we do not ... yet. Error out if we find them.
+   ret = nf90_inquire_attribute(ncfile_in, var_id, 'scale_factor')
+   if (ret == NF90_NOERR) failure = .true.
+
+   ret = nf90_inquire_attribute(ncfile_in, var_id, 'offset')
+   if (ret == NF90_NOERR) failure = .true.
+
+   if (failure) then
+      write(msgstring2,*)'the use of scale_factor and/or offset is not supported.'
+      call error_handler(E_ERR,routine,varname, source, text2=msgstring2)
+   endif
+
+   ! Finally read the variable.
```

Comment:
By not providing the 'start' optional variable to nf90_get_var, it defaults to an array of ones ... so it is always starting at timestep 1, for example ... and using the dimensions (dims) from 'get_io_dim_lengths()' ...

Reply:
sorry, i don't understand this comment.  is it for pointing out where future work might need to happen? the nf90_get_var() call on line 899 hasn't changed from the original.